### DISCUSSION TITLE
[Question] How to change font style inside text editor?

### DISCUSSION THREAD

**Discussion by tamasregenye** [hm21/pro_image_editor#264] [Q&A] [answered]
There is a `customTextStyles `property inside `textEditorConfigs`, which is a List. However, I couldn't find any button in the editor that allows switching between the custom text styles.

This is an awesome package but this feature is a must-have for my use case. Could someone guide me on how to enable or implement this functionality?

Thanks in advance!

**Comment by hm21** [comment, marked as answer]
To enable multiple text styles in the text editor, include `customTextStyles` in the `textEditorConfigs` and set `showSelectFontStyleBottomBar` to `true`. This will display a bottom bar, allowing you to choose between various custom text styles.

I recommend you also to check out the [web demo](https://hm21.github.io/pro_image_editor) where you can test the [Google-Font-Emojis example](https://github.com/hm21/pro_image_editor/blob/stable/example/lib/pages/google_font_example.dart), which demonstrates the use of multiple Google Fonts within the text editor.

**Reply by justAsascha** [reply]
I need a custom implementation of the font selection (different UI/UX). 
Is there a (better) programmatically way? 

```dart
void changeFont(TextEditorState editor, String fontFamily) {
    editor.selectedTextStyle = editor.selectedTextStyle
        .copyWith(fontFamily: fontFamily);
    // hack: trigger rendering of current text span
    final oldText = editor.textCtrl.text;
    editor.textCtrl.text = "";
    editor.textCtrl.text = oldText;
  }
```
The only way I found to trigger a rendering of the textSpan is to update the text value.

**Reply by hm21** [reply]
I recommend using the `setTextStyle` method to update the `fontStyle` rather than directly modifying `selectedTextStyle`. Directly editing `selectedTextStyle` is only necessary in rare cases where a completely custom UI update logic is required.

**Example:**

```dart
void changeFont(TextEditorState editor, String fontFamily) {
  editor.setTextStyle(
    editor.selectedTextStyle.copyWith(
      fontFamily: fontFamily,
    ),
  );
}
```
